### PR TITLE
OBSDOCS-937/TRACING-4071: Write documentation for oidcauthextension component

### PR DESCRIPTION
**Pull request by max-cx**
:warning: This PR is tied to a product release date, which is currently set to **June 5** (Wednesday).

- If the merge review gets delayed, you're welcome to ping `mleonov` on Slack.
- If you have any questions, ping `mleonov` on Slack.

Signed-off-by: Andreas Gerstmayr <agerstmayr@redhat.com>

Edit of https://github.com/openshift/openshift-docs/pull/73780

Version(s):

4.12, 4.13, 4.14, 4.15, 4.16

Issue:

https://issues.redhat.com/browse/OBSDOCS-937
https://issues.redhat.com/browse/TRACING-4071

Link to docs preview:

https://75487--ocpdocs-pr.netlify.app/openshift-enterprise/latest/observability/otel/otel-configuration-of-otel-collector.html#oidcauth-extension_otel-configuration-of-otel-collector

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by IshwarKanse — APPROVED** [review]

**Review by andreasgerstmayr — APPROVED** [review]

File: modules/otel-collector-components.adoc [hunk 1]
```diff
@@ -871,6 +871,48 @@ This extension supports traces, metrics, and logs.
...
+    service:
+      extensions: [oidc]
+      pipelines:
+        traces:
+          receivers: [otlp]
+          exporters: [otlp]
+----
+<1> The name of the header that contains the ID token. The default name is `authorization`.
+<2> The base URL of the OIDC provider.
+<3> This field is optional. The path to the issuer's CA certificate.
```
Comment (kowen-rh) [inline modules/otel-collector-components.adoc:912]:
```adoc
<3> (Optional) The path to the issuer's CA certificate.
```

**Review by kowen-rh — COMMENTED** [review]
Just a couple of minor notes/questions. :slightly_smiling_face: 

/remove-label peer-review-in-progress
/label peer-review-done

File: modules/otel-collector-components.adoc [hunk 2]
```diff
@@ -871,6 +871,48 @@ This extension supports traces, metrics, and logs.
...
+<4> The audience for the token.
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-collector-components.adoc:913]: @kowen-rh, oh hello, Kevin, long time no see! I'm just getting to this PR.

Thank you.

I rewrote it as follows:

`<3> Optional: The path to the issuer's CA certificate.`

File: modules/otel-collector-components.adoc (hunk 1 above)
Comment (max-cx) [reply modules/otel-collector-components.adoc:912]: @kowen-rh, oh hello, Kevin, long time no see! I'm just getting to this PR.

Thank you.

I rewrote it as follows:

`<3> Optional: The path to the issuer's CA certificate.`

**Comment by max-cx** [comment]
Closing this PR because I cherry-picked its commit into https://github.com/openshift/openshift-docs/pull/76340.

**Review by michaelryanpeter — COMMENTED** [review]
minor nit that is not a blocker

/label ok-to-merge
/label do-not-merge
/remove-label merge-review-in-progress
/remove-label merge-review-needed

File: modules/otel-collector-components.adoc [hunk 3]
```diff
@@ -871,6 +871,48 @@ This extension supports traces, metrics, and logs.
 <12> You can assign the authenticator configuration to an OTLP exporter.
 
 
+[id="oidcauth-extension_{context}"]
+=== OIDC Auth Extension
+
+:FeatureName: The OIDC Auth Extension
+include::snippets/technology-preview.adoc[]
```
[the remark this answers is not in the document]
Comment (max-cx) [reply modules/otel-collector-components.adoc:878]: Thank you. I plan to open a global PR for this update throughout this file.

**Pull request merged**